tool-plugins/intellij/src/main/java/io/ballerina/plugins/idea/preloading/BallerinaEditorFactoryListener.java

```diff
@@ -0,0 +1,69 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package io.ballerina.plugins.idea.preloading;
+
+import com.intellij.openapi.editor.event.EditorFactoryEvent;
+import com.intellij.openapi.editor.event.EditorFactoryListener;
+import com.intellij.openapi.fileEditor.FileDocumentManager;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.testFramework.LightVirtualFileBase;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import static io.ballerina.plugins.idea.BallerinaConstants.BAL_FILE_EXT;
+import static io.ballerina.plugins.idea.preloading.LSPUtils.registerServerDefinition;
+
+/**
+ * Editor listener implementation which is used to handle ballerina source files inside non-ballerina projects.
+ *
+ * @since 1.1.4
+ */
+public class BallerinaEditorFactoryListener implements EditorFactoryListener {
+
+    private Project project;
+    private boolean balSourcesFound = false;
+
+    public BallerinaEditorFactoryListener(Project project) {
+        this.project = project;
+    }
+
+    @Override
+    public void editorCreated(@NotNull EditorFactoryEvent event) {
+        Project project = event.getEditor().getProject();
+        if (project == null) {
+            return;
+        }
+        VirtualFile file = FileDocumentManager.getInstance().getFile(event.getEditor().getDocument());
+        if (!balSourcesFound && project.equals(this.project) && isBalFile(file)) {
+            registerServerDefinition(project);
+            balSourcesFound = true;
+        }
+    }
+
+    public static boolean isBalFile(@Nullable VirtualFile file) {
```

Comment:
Shall we document these methods since it a public API?

Reply:
This supposed to be a private util method and corrected accordingly. Thanks for pointing that out.